docs/development/module/fees.md

```diff
@@ -0,0 +1,209 @@
+can be dynamic and based on what `WeightToFee` is defined to be, the final fee can have some degree
+of variability. As for the length fee, the inputs of the transaction could change the length and
+hence affecting the length fee. Nonetheless, these changes are independent and a _general update
+logic to the **entire fee** cannot be composed out of them trivially_. In other words, for any
+dispatchable, given the same inputs, _it will always incur the same cost_. This might not always be
+desirable. Chains might need to increase or decrease fees based on some condition.  To fulfill this
+requirement, Substrate provides:
+  - a multiplier stored in the transaction-payment module that is applied to the outcome of the
+    above formula by default (needless to say, the default value of which is _multiplication
+    identity_, meaning that it has no effect). This is stored in
+    [`NextFeeMultiplier`](/rustdocs/master/srml_transaction_payment/struct.Module.html#method.next_fee_multiplier)
+    storage and can be configured through the genesis spec of the module.
+  - a configurable parameter for a runtime to describe how this multiplier can change. This is
+    expressed via
+    [`FeeMultiplierUpdate`](/rustdocs/master/srml_transaction_payment/trait.Trait.html#associatedtype.FeeMultiplierUpdate).
+
+`NextFeeMultiplier` has the type `Fixed64`, which can represent a fixed point number with a billion
+points accuracy. So, given the final fee formula above, the final version would be:
+
+```
+fee =
+  base_fee +
+  len(tx) * length_fee +
+  WeightToFee(weight)
+
+final_fee = fee * NextFeeMultiplier
+```
+
+Updating the `NextFeeMultiplier` has a similar manner as `WeightToFee`. The
+[`FeeMultiplierUpdate`](/rustdocs/master/srml_transaction_payment/trait.Trait.html#associatedtype.FeeMultiplierUpdate)
+associated type in `transaction-payment` is defined as a `Convert<Fixed64, Fixed64>`, which should
+be read as: it receives the previous multiplier and spits out the next one.
+
+The default update function is inspired by the Polkadot network and implements a targeted adjustment
+in which a target saturation level of block weight is defined. If the previous block is more
+saturated, then the fees are slightly increases. Similarly, if less transaction than the target are
+in the previous block, fees are decreased by a small amount. More information about this can be
+found in the [web3 research
+page](https://research.web3.foundation/en/latest/polkadot/Token%20Economics/#relay-chain-transaction-fees).
+
+## Substrate Weight System
+
+> This section assumes that you have already read the `Weight` section
+
+The entire substrate runtime module library is already annotated with a simple and fixed weight
+system. A user can decide to use the same system or implement a new one from scratch. The latter is
+outside the scope of this document and is explained int he dedicated [`Weight`]() conceptual
```

Comment:
I think we should bring your explanation into this document, since it makes sense as a developer doc.